pkg/util/ovs/fake_ovs.go

```diff
@@ -200,52 +202,78 @@ func fixFlowFields(flow *OvsFlow) {
 }
 
 func (tx *ovsFakeTx) AddFlow(flow string, args ...interface{}) {
-	if tx.err != nil {
-		return
+	if len(args) > 0 {
+		flow = fmt.Sprintf(flow, args...)
 	}
-	parsed, err := ParseFlow(ParseForAdd, flow, args...)
+	tx.flows = append(tx.flows, fmt.Sprintf("add %s", flow))
+}
+
+func (fake *ovsFake) addFlowHelper(flow string) error {
+	parsed, err := ParseFlow(ParseForAdd, flow)
 	if err != nil {
-		tx.err = err
-		return
+		return err
 	}
 	fixFlowFields(parsed)
 
 	// If there is already an exact match for this flow, then the new flow replaces it.
-	for i := range tx.fake.flows {
-		if FlowMatches(&tx.fake.flows[i], parsed) {
-			tx.fake.flows[i] = *parsed
-			return
+	for i := range fake.flows {
+		if FlowMatches(&fake.flows[i], parsed) {
+			fake.flows[i] = *parsed
+			return nil
 		}
 	}
 
-	tx.fake.flows = append(tx.fake.flows, *parsed)
-	sort.Sort(ovsFlows(tx.fake.flows))
+	fake.flows = append(fake.flows, *parsed)
+	sort.Sort(ovsFlows(fake.flows))
+	return nil
 }
 
 func (tx *ovsFakeTx) DeleteFlows(flow string, args ...interface{}) {
-	if tx.err != nil {
-		return
+	if len(args) > 0 {
+		flow = fmt.Sprintf(flow, args...)
 	}
-	parsed, err := ParseFlow(ParseForFilter, flow, args...)
+	tx.flows = append(tx.flows, fmt.Sprintf("delete %s", flow))
+}
+
+func (fake *ovsFake) deleteFlowsHelper(flow string) error {
+	parsed, err := ParseFlow(ParseForFilter, flow)
 	if err != nil {
-		tx.err = err
-		return
+		return err
 	}
 	fixFlowFields(parsed)
 
-	newFlows := make([]OvsFlow, 0, len(tx.fake.flows))
-	for _, flow := range tx.fake.flows {
+	newFlows := make([]OvsFlow, 0, len(fake.flows))
+	for _, flow := range fake.flows {
 		if !FlowMatches(&flow, parsed) {
 			newFlows = append(newFlows, flow)
 		}
 	}
-	tx.fake.flows = newFlows
+	fake.flows = newFlows
+	return nil
 }
 
-func (tx *ovsFakeTx) EndTransaction() error {
-	err := tx.err
-	tx.err = nil
-	return err
+func (tx *ovsFakeTx) Commit() error {
```

Comment:
This doesn't actually implement transactional behavior: if you do

    tx.AddFlow("table=10, actions=drop")
    tx.AddFlow("@!#$!@#$!@#$")
    tx.Commit()

then it will return an error saying that the second flow couldn't be parsed, but the `"table=10, actions=drop"` flow will still have been added.

There should probably be a test of transactionality in `fake_ovs_test.go`